### DISCUSSION TITLE
Get and Set alphaMode/alphaCutoff or doubleSided

### DISCUSSION THREAD

**Discussion by shamsail** [google/model-viewer#2558] [Q&A]
Hi, Currently I'm working on the manipulation of the materials and textures. 
I'm having an issue regarding the getter and setter of 'alphaMode'.
What I'm experiencing is that whenever I change the confings of material, it's changing the alphaMode to "OPAQUE" whether the material has alphaMode of "OPAQUE", "BLEND" , "MASK".

Main issue is coming when I'm manipulating the material that have alphaMode besides "OPAQUE", Like in my case, I have a material in the model that is glass and has "BLEND" alphaMode. But It's blendMode is changing to "OPAQUE" as soon as I select this glassy material and make some configs of matalic/roughness and glassy material looses it's transparency.

I'm seeing through console that there is no getter and setter for alphaMode. 
Meanwhile, I'm able to get the alphaMode through 'sourceObject' symbol but there is no way to set the alphaMode again.

Is there any way to get and set the alphaMode of material?
Also, the getter and setter for alphaCutOff and for doubleSided

I was exploring the git repository and There were some functions like
alphaMode() and setAlphaMode , alphaCutOff and setAlphaCurOff, doubleSided and setDoubleSided and so on.. BUT these are not being exposed in the scene graph API.

![Screenshot from 2021-07-06 17-14-51](https://user-images.githubusercontent.com/30554980/124604508-71217700-de84-11eb-9a46-1b623ed6df99.png)
![Screenshot from 2021-07-06 17-15-39](https://user-images.githubusercontent.com/30554980/124604539-78e11b80-de84-11eb-9f5b-a159ef17d261.png)
![Screenshot from 2021-07-06 17-16-02](https://user-images.githubusercontent.com/30554980/124604540-7979b200-de84-11eb-8fe5-c1381afd6218.png)

Is there any way to handle this?

**Edited:**
My issue has been resolved
```
Main issue is coming when I'm manipulating the material that have alphaMode besides "OPAQUE", Like in my case, I have a material in the model that is glass and has "BLEND" alphaMode. But It's blendMode is changing to "OPAQUE" as soon as I select this glassy material and make some configs of matalic/roughness and glassy material looses it's transparency.
```
It was error on my side. As, I was missing the alpha channel of **baseColorFactor** rgb**a** . It was always set to '1' that was effecting tranparency.

baseColorFactor -> rgba
emissiveFactor -> rgb

But still I want to know the getter and setter for alphaMode/alphaCutOff/doubleSided..
I think and believe, it will be available in the next release of the package.

**Comment by elalish** [comment]
Yes, we need to implement these. FYI @timmmeh

**Comment by ghost** [comment]
Got it, added to the list
